pilot/pkg/networking/core/v1alpha3/loadbalancer/loadbalancer.go

```diff
@@ -39,7 +41,8 @@ func ApplyLocalityLBSetting(
 	// one of Distribute or Failover settings can be applied.
 	if localityLB.GetDistribute() != nil {
 		applyLocalityWeight(locality, loadAssignment, localityLB.GetDistribute())
-	} else {
+	} else if outlierDetection != nil {
+		// Failover needs outlier detection, otherwise Envoy will never drop down to a lower priority.
```

Comment:
I don't think a warning here is a good idea, this is called for each push, and it isn't incorrect to not have an outlier detection set. This is enabled by default, so that would mean we get a warning that outlierDetection isn't set for bookinfo, etc, out of the box which would likely just cause more confusion.

If locality LB was an option that can be turned on or off then it would make sense, but given it isn't I don't think it makes sense to warn here.